table/block_based/block_based_table_reader.cc

```diff
@@ -379,13 +379,22 @@ void BlockBasedTable::UpdateCacheInsertionMetrics(
   }
 }
 
+template <typename TBlocklike>
 Cache::Handle* BlockBasedTable::GetEntryFromCache(
     const CacheTier& cache_tier, Cache* block_cache, const Slice& key,
     BlockType block_type, const bool wait, GetContext* get_context,
     const Cache::CacheItemHelper* cache_helper,
-    const Cache::CreateCallback& create_cb, Cache::Priority priority) const {
+    Cache::Priority priority) const {
   Cache::Handle* cache_handle = nullptr;
   if (cache_tier == CacheTier::kNonVolatileBlockTier) {
```

Comment:
This doesn't actually indicate whether a secondary cache is configured, right? I think you'll need to add something to Cache to expose that (and likely save it to avoid repeated calls).

Reply:
right, just realized that the default tier is set to `kNonVolatileBlockTier`, and it doesn't indicate it's secondarycache.
I updated it to capture less variables so it won't need to allocate extra memory from heap.